### PR TITLE
Sitipe plugin

### PR DESCRIPTION
**Pull request by pascalwilbrink**

### CODE REVIEW

File: public/js/plugins.js [hunk 1]
```diff
@@ -319,5 +319,12 @@ export const officialPlugins = [
     kind: 'menu',
     requireDoc: true,
     position: 'middle'
-  }
+  },
+  {
+    name: 'Sitipe',
+    src: '/src/editors/Sitipe.js',
+    icon: 'developer_board',
```
Comment (JakobVogelsang) [inline public/js/plugins.js:326]:
```suggestion
    icon: 'conveyor_belt',
```

Comment (JakobVogelsang) [inline public/js/plugins.js:326]: This icon is strongly connected with IED in CoMPAS. Sorry we did not specify this at all. I had a look for another icon and made a proposal. What do you think about this?

File: src/editors/Sitipe.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import { css, html, LitElement, property, TemplateResult } from 'lit-element';
+import { translate } from 'lit-translate';
+
+import './sitipe/sitipe-substation.js';
+import { isPublic } from '../foundation.js';
+
+/** An editor [[`plugin`]] for Subscribe Data Binding (SMV). */
```
Comment (JakobVogelsang) [inline src/editors/Sitipe.ts:7]: Copy leaftover?

File: src/editors/sitipe/foundation.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+/** `Private`-safeguarded selectors for `Substation` and its descendants */
+export const selectors = <Record<SubstationTag, string>>(
+  Object.fromEntries(
+    substationPath.map((e, i, a) => [e, a.slice(0, i + 1).join(' > ')])
+  )
+);
+
+export const SIEMENS_SITIPE_IED = 'Siemens-SITIPE-IEDRef';
+
+export const SIEMENS_SITIPE_TEMPLATE = 'Siemens-SITIPE-BayTemplate';
```
Comment (JakobVogelsang) [inline src/editors/sitipe/foundation.ts:15]: I would be more precise on that. It is a bay template. In SIPITE there are a lot of others as well.

File: src/editors/sitipe/sitipe-substation.ts [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+import '../../action-pane.js';
+import '../../action-icon.js';
+
+import {
+  selectors,
+  SIEMENS_SITIPE_IED,
+  SIEMENS_SITIPE_TEMPLATE,
+} from './foundation.js';
+
+/** [[`Substation`]] plugin subeditor for editing `Substation` sections. */
```
Comment (JakobVogelsang) [inline src/editors/sitipe/sitipe-substation.ts:20]: Copy leftover?

File: src/editors/sitipe/sitipe-substation.ts [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+@customElement('sitipe-substation')
+export class SitipeSubstation extends LitElement {
+  /** The document being edited as provided to editor by [[`Zeroline`]]. */
```
Comment (JakobVogelsang) [inline src/editors/sitipe/sitipe-substation.ts:23]: Copy leftover?

File: src/editors/sitipe/sitipe-substation.ts [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+    const template: string =
+      bay.querySelector(`Private[type="${SIEMENS_SITIPE_TEMPLATE}"]`)
+        ?.textContent ?? '';
+
+    return html`
+      <div>
+        ${Array.from(
+          bay.querySelectorAll(`Private[type="${SIEMENS_SITIPE_IED}"]` ?? [])
+        ).map(
+          ied =>
```
Comment (JakobVogelsang) [inline src/editors/sitipe/sitipe-substation.ts:75]: How about `iedTempalte`

File: src/editors/sitipe/sitipe-substation.ts [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+      <div class="bayContainer">
+        ${Array.from(voltageLevel.querySelectorAll(selectors.Bay) ?? []).map(
+          this.renderBay.bind(this)
+        )}
+      </div>
+    </action-pane>`;
+  }
+
+  render(): TemplateResult {
+    return html`<oscd-action-pane label="${this.substationHeader}">
```
Comment (JakobVogelsang) [inline src/editors/sitipe/sitipe-substation.ts:104]: Why do we use `oscd-action-pane` and `action-pane` in the same file?

**Review by JakobVogelsang — CHANGES_REQUESTED** [review]
Mostly minor thing other than that LGTM

File: src/editors/sitipe/sitipe-substation.ts (hunk 7 above)
Comment (pascalwilbrink) [reply src/editors/sitipe/sitipe-substation.ts:104]: Fixed

File: src/editors/sitipe/sitipe-substation.ts (hunk 6 above)
Comment (pascalwilbrink) [reply src/editors/sitipe/sitipe-substation.ts:75]: fixed.

change the icon to conveyor_belt as well?

File: src/editors/sitipe/sitipe-substation.ts (hunk 5 above)
Comment (pascalwilbrink) [reply src/editors/sitipe/sitipe-substation.ts:23]: fixed

File: src/editors/sitipe/sitipe-substation.ts (hunk 4 above)
Comment (pascalwilbrink) [reply src/editors/sitipe/sitipe-substation.ts:20]: fixed

File: src/editors/sitipe/foundation.ts (hunk 3 above)
Comment (pascalwilbrink) [reply src/editors/sitipe/foundation.ts:15]: Fixed. 

SIEMENS_SITIPE_IED_REF
and
SIEMENS_SITIPE_BAY_TEMPLATE

File: src/editors/Sitipe.ts (hunk 2 above)
Comment (pascalwilbrink) [reply src/editors/Sitipe.ts:7]: fixed

File: public/js/plugins.js (hunk 1 above)
Comment (pascalwilbrink) [reply public/js/plugins.js:326]: fixed

File: src/editors/sitipe/sitipe-substation.ts (hunk 6 above)
Comment (JakobVogelsang) [reply src/editors/sitipe/sitipe-substation.ts]: You mean in the action-icon? Then no. Those are showing an IED, and therefore I would say the icon `developer_board` is fine. :)

Comment (pascalwilbrink) [reply src/editors/sitipe/sitipe-substation.ts]: Fixed

**Review by JakobVogelsang — APPROVED** [review]
LGTM

**Pull request merged**